WcaOnRails/app/views/registrations/register.html.erb

```diff
@@ -32,7 +32,8 @@
     <% end %>
   <% else %>
     <% user_may_register = current_user.cannot_register_for_competition_reasons.length == 0 %>
-    <% if @registration.show_details? %>
+    <% current_user_can_pre_register = @competition.user_can_pre_register?(current_user) %>
+    <% if @registration.show_details? || current_user_can_pre_register %>
```

Comment:
Why do you need this additional `OR` check here?
If it is absolutely necessary, can you perhaps merge it into the `show_details?` method? Or will that have too many side effects?

Reply:
Yeah, the show_details method is only really used for this purpose anyway so I have gone ahead and merged them.